### PR TITLE
ISPN-7129 Duplicate cache names shouldn't be allowed

### PR DESCRIPTION
**Pull request by jmarkos**
Reopening https://github.com/infinispan/infinispan/pull/4631 (rebased).

### CODE REVIEW

File: core/src/main/java/org/infinispan/configuration/parsing/Parser.java [hunk 1]
```diff
@@ -2159,20 +2159,20 @@ private void parseGroups(final XMLExtendedStreamReader reader, final Configurati
...
-            ConfigurationBuilder baseConfigurationBuilder = holder.getNamedConfigurationBuilders().get(baseConfigurationName);
-            if (baseConfigurationBuilder == null) {
-               throw log.undeclaredConfiguration(baseConfigurationName, name);
-            }
-            Configuration baseConfiguration = baseConfigurationBuilder.build();
-            if (!baseConfiguration.isTemplate()) {
-               throw log.noConfiguration(baseConfigurationName);
-            }
-            builder.read(baseConfiguration);
+      if (holder.getNamedConfigurationBuilders().get(name) != null) {
```
Comment (tristantarrant) [inline core/src/main/java/org/infinispan/configuration/parsing/Parser.java]: I'd rather use containsKey

Comment (jmarkos) [inline core/src/main/java/org/infinispan/configuration/parsing/Parser.java]: done

**Comment by jmarkos** [comment]
Rebased, updated.

**Comment by tristantarrant** [comment]
Pushed to master, thanks

**Pull request merged**